ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/MappedOperationPool.java

```diff
@@ -89,6 +100,52 @@ public MappedOperationPool(
             OPERATION_POOL_SIZE_VALIDATION_REASON + metricType,
             "Total number of attempts to add an operation to the pool, broken down by validation result",
             "result");
+
+    this.timeProvider = timeProvider;
+    asyncRunner.runWithFixedDelay(
+        this::updateLocalSubmissions,
+        Duration.ofMinutes(10),
+        this::updateLocalSubmissionsErrorHandler);
+  }
+
+  private void updateLocalSubmissionsErrorHandler(Throwable throwable) {}
+
+  private void updateLocalSubmissions() {
+    final UInt64 staleTime =
+        timeProvider.getTimeInSeconds().minus(Duration.ofHours(2).getSeconds());
+    final List<OperationPoolEntry<T>> staleLocalOperations =
+        operations.values().stream()
+            .filter(OperationPoolEntry::isLocal)
+            .filter(entry -> entry.getTimeSubmitted().isLessThanOrEqualTo(staleTime))
+            .collect(Collectors.toList());
+    if (!staleLocalOperations.isEmpty()) {
+      LOG.info(
+          "Re-publishing {} operations that are still in the local {} operation pool",
+          staleLocalOperations.size(),
+          metricType);
+    }
+    int i = 0;
+    for (OperationPoolEntry<T> entry : staleLocalOperations) {
+      if (++i % 200 == 0) {
+        try {
+          // if we're processing large numbers, don't flood the network...
+          Thread.sleep(100);
```

Comment:
Just confirming: this will sleep the thread from `operationPoolAsyncRunner`, right?

Reply:
yep will sleep the async runner, we have 2 threads.. .and it'll only happen if we get more than 200 entries...

Reply:
what about putting those constants in a `private static` var?